doc/coding.rst

```diff
@@ -0,0 +1,262 @@
+Coding
+======
+
+Coding is done from within your ``virtualenv``, ideally on one of the 904 DAQ
+machines; ``lxplus`` is only supported on a *best-effort* basis. Follow the
+:doc:`setup instructions <setup>` first (or :doc:`Setting up a virtualenv at P5
+<virtualenv-p5>` if you
+need to work there). It is recommended that you use a separate ``virtualenv``
+for development. The instructions below assume that your ``virtualenv`` is
+installed and activated.
+
+First, install the dependencies for development:
+
+.. code-block:: bash
+
+    pip install -r requirements-dev.txt
+
+.. note::
+
+    In case ``bash`` cannot find ``pip``, try ``python -m pip ...`` instead.
+
+Clone the repository and move to the code directory:
+
+.. code-block:: bash
+
+    git clone git@github.com:cms-gem-daq-project/gem-plotting-tools.git
+    cd gem-plotting-tools
+
+Then, you need to import some code from the `gembuild`_ project:
+
+.. code-block:: bash
+
+    git submodule init
+    git submodule update
+
+This enable the use of the ``Makefile``.
+
+Building and installing
+-----------------------
+
+In order to test your code, you need to install it inside your ``virtualenv``.
+It is done by building a ``pip`` package:
+
+.. code-block:: bash
+
+    make pip
+
+.. note::
+
+    In case you encounter strange build errors, try to run the following command
+    first:
+
+    .. code-block:: bash
+
+        git submodule update
+
+    This will update your local version of `gembuild`_.
+
+Once the package is built, it can be installed using:
+
+.. code-block:: bash
+
+    pip install -U rpm/gempython_gemplotting-1.0.0.tar.gz
```

Comment:
The developer `gist` has this done with option `-I` to ignore and force re-install.  In some cases I've found that the `-U` does not always overwrite correctly previously installed files.  Would suggest to have `-I`.